app/security_system.py

```diff
@@ -72,13 +73,43 @@ def recognize_face(self, frame):
 
         # TODO 3
         # Perform face detection
+        faces = face_cascade.detectMultiScale(gray_frame, scaleFactor = 1.2, minNeighbors = 5, minSize = (30,30))
         # Iterate through each detected face
-        # Extract the detected face region
-        # Perform facial recognition on the detected face
-        # Draw bounding box around the detected face
-        # Annotate the frame with the recognized user's name and authorization status
-        # Log access attempt
-        # Your code goes here
+        for face in faces:
+            # Extract the detected face region
+            x,y,w,h = face
+            X,Y = x+w,y+h
+            image = gray_frame[y:Y, x:X]
+            # Perform facial recognition on the detected face
+            label, confidence = self.face_recognizer.predict(image)
+            person_name = self.get_person_name(label,confidence)
+            authorization_status = self.is_person_authorized(person_name)
+            if(authorization_status==True):
+                color = (0,200,100)
+                is_authorized = "True"
+            else:
+                color = (0,35,200)
+                is_authorized = "False"
+            # Draw bounding box around the detected face
+            l= 20
+            t= 5
```

Comment:
Variable `thickness` would make code readable that `t`.

Reply:
That is true. I'll edit that in my next commit.